### PR TITLE
Move TimeBudget to vespajlib and use Clock

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

**Comment by hakonhall** [comment]
FYI @vekterli

File: orchestrator/src/test/java/com/yahoo/vespa/orchestrator/resources/HostResourceTest.java [hunk 1]
```diff
@@ -61,7 +62,7 @@
  * @author hakonhall
  */
 public class HostResourceTest {
-    private static final Timer timer = mock(Timer.class);
+    private static final Clock clock = mock(Clock.class);
```
Comment (bratseth) [inline orchestrator/src/test/java/com/yahoo/vespa/orchestrator/resources/HostResourceTest.java]: No need to use a mock framework here. You could use e.g com.yahoo.test.ManualClock instead.

File: vespajlib/src/main/java/com/yahoo/time/UncheckedTimeoutException.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * Exception thrown when a blocking operation times out.
+ *
+ * <p>Mirrors {@link java.util.concurrent.TimeoutException}
+ *
+ * @author hakon
+ */
+@SuppressWarnings("serial")
+public class UncheckedTimeoutException extends RuntimeException {
```
Comment (bratseth) [inline vespajlib/src/main/java/com/yahoo/time/UncheckedTimeoutException.java]: I suggest removing the "Unchecked" prefix. It is used when there is already a checked variant in the same package, but there is not here.

File: vespajlib/src/test/java/com/yahoo/time/TimeBudgetTestCase.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import java.time.Duration;
+import java.time.Instant;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class TimeBudgetTestCase {
+    private final Clock clock = mock(Clock.class);
```
Comment (bratseth) [inline vespajlib/src/test/java/com/yahoo/time/TimeBudgetTestCase.java]: Same, please don't mock frivolously.

**Review by bratseth — APPROVED** [review]
LGTM, consider my comments.

File: vespajlib/src/main/java/com/yahoo/time/UncheckedTimeoutException.java (hunk 2 above)
Comment (mpolden) [reply vespajlib/src/main/java/com/yahoo/time/UncheckedTimeoutException.java]: There's also `com.google.common.util.concurrent.UncheckedTimeoutException` from Guava, which is used in several places already.

File: orchestrator/src/test/java/com/yahoo/vespa/orchestrator/resources/HostResourceTest.java (hunk 1 above)
Comment (hakonhall) [reply orchestrator/src/test/java/com/yahoo/vespa/orchestrator/resources/HostResourceTest.java]: Didn't know about the ManualClock.

File: vespajlib/src/main/java/com/yahoo/time/UncheckedTimeoutException.java (hunk 2 above)
Comment (hakonhall) [reply vespajlib/src/main/java/com/yahoo/time/UncheckedTimeoutException.java]: OK, done.

File: orchestrator/src/test/java/com/yahoo/vespa/orchestrator/resources/HostResourceTest.java (hunk 1 above)
Comment (hakonhall) [reply orchestrator/src/test/java/com/yahoo/vespa/orchestrator/resources/HostResourceTest.java]: ManualClock would introduce an additional dependency here, and no manipulation of the clock is necessary - leaving as-is. Will move to ManualClock in TimeBudgetTestCase.java

File: vespajlib/src/test/java/com/yahoo/time/TimeBudgetTestCase.java (hunk 3 above)
Comment (hakonhall) [reply vespajlib/src/test/java/com/yahoo/time/TimeBudgetTestCase.java]: Done

File: vespajlib/src/main/java/com/yahoo/time/UncheckedTimeoutException.java (hunk 2 above)
Comment (hakonhall) [reply vespajlib/src/main/java/com/yahoo/time/UncheckedTimeoutException.java]: For some reason github didn't show Martin's comment before now. Moving to guava.

**Pull request merged**